android/src/main/java/com/reactnativecommunity/webview/RNCWebViewModule.java

```diff
@@ -50,6 +55,35 @@
   private File outputVideo;
   private DownloadManager.Request downloadRequest;
 
+  protected static class ShouldOverrideUrlLoadingLock {
+    protected enum ShouldOverrideCallbackState {
+      UNDECIDED,
+      SHOULD_OVERRIDE,
+      DO_NOT_OVERRIDE,
+    }
+
+    private final Random randomGenerator = new Random();
+    private final HashMap<Integer, AtomicReference<ShouldOverrideCallbackState>> shouldOverrideLocks = new HashMap<>();
+
+    public synchronized Pair<Integer, AtomicReference<ShouldOverrideCallbackState>> getNewLock() {
+      Integer nextRand = randomGenerator.nextInt();
```

Comment:
random could have collisions. Replace random number generator with a static AtomicInteger and just increment the counter.

Reply:
The possibility of a collision seems extremely remote. I could do an incrementing integer, but didn't want to think about overflow behavior too much (though overflow on its face doesn't seem like an issue).

I don't believe an `AtomicInteger` would be necessary though, as we are `synchronized` on `getNewLock()`.